guppylang/std/qsystem/__init__.py

```diff
@@ -74,6 +76,42 @@ def reset(q: qubit) -> None: ...
 def qfree(q: qubit @ owned) -> None: ...
 
 
+@guppy.hugr_op(quantum_op("LazyMeasureLeaked", ext=QSYSTEM_EXTENSION))
+@no_type_check
+def _measure_leaked(q: qubit @ owned) -> Future[int]:
+    """Measure the quibit or return 2 if it is leaked."""
+
+
+@guppy
+def measure_leaked(q: qubit @ owned) -> "MaybeLeaked":
+    fm = _measure_leaked(q)
+    return MaybeLeaked(fm)
+
+@guppy
+class MaybeLeaked:
```

Comment:
This should be a struct
```suggestion
@guppy.struct
class MaybeLeaked:
```